src/index.js

```diff
@@ -13,33 +13,39 @@ export default function Composer({
 
   /**
    * Recursively build up elements from props.components and accumulate `results` along the way.
-   * @param next The first entry in components
-   * @param ...remaining The remaining entries in components
-   * @param results This is the argument that we pass into `children`
+   * @param {Array.<ReactElement|Function>} components
+   * @param {Array} results
    * @returns {ReactElement}
    */
-  function chainComponents([next, ...remaining], results = []) {
```

Comment:
:memo: For the record: dropping this `[next, ...remaining]` in favor of `components[0]`/`components.slice(1)` did make me feel quite sad 😢 😆

Reply:
One day... :)